src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -191,37 +191,21 @@ def variant_compatible_siblings(filenames, variant=None) -> Union[List[os.PathLi
     # all variant filenames will be used by default
     usable_filenames = set(variant_filenames)
 
-    def convert_to_variant(filename):
-        if "index" in filename:
-            variant_filename = filename.replace("index", f"index.{variant}")
-        elif re.compile(f"^(.*?){transformers_index_format}").match(filename) is not None:
-            variant_filename = f"{filename.split('-')[0]}.{variant}-{'-'.join(filename.split('-')[1:])}"
-        else:
-            variant_filename = f"{filename.split('.')[0]}.{variant}.{filename.split('.')[1]}"
-        return variant_filename
-
     def find_component(filename):
         if not len(filename.split("/")) == 2:
             return
         component = filename.split("/")[0]
         return component
 
-    def has_sharded_variant(component, variant, variant_filenames):
-        # If component exists check for sharded variant index filename
-        # If component doesn't exist check main dir for sharded variant index filename
+    def has_variant(filename, variant_filenames):
+        component = find_component(filename)
         component = component + "/" if component else ""
-        variant_index_re = re.compile(
-            rf"{component}({'|'.join(weight_prefixes)})\.({'|'.join(weight_suffixs)})\.index\.{variant}\.json$"
-        )
-        return any(f for f in variant_filenames if variant_index_re.match(f) is not None)
 
-    for filename in non_variant_filenames:
-        if convert_to_variant(filename) in variant_filenames:
-            continue
+        return any(f.startswith(component) for f in variant_filenames)
```

Comment:
what appends if like we only have a bf16.bin and this is a non-variant safetensors ? stuff like this

Reply:
As in we are trying to fetch something like this? 

```
        variant = "fp16"
        filenames = [
                f"vae/diffusion_pytorch_model.{variant}.bin",
                f"text_encoder/model.{variant}.bin",
                f"unet/diffusion_pytorch_model.{variant}.bin",
        ]
        model_filenames, variant_filenames = variant_compatible_siblings(filenames, variant=None)
```

Reply:
like this, I think we should fetch the non-variant safetensors in this case, no?         
        
        variant = "fp16"
        filenames = [
                f"vae/diffusion_pytorch_model.{variant}.bin",
                f"text_encoder/model.{variant}.bin",
                f"unet/diffusion_pytorch_model.{variant}.bin",
                f"vae/diffusion_pytorch_model.safetensors",
                f"text_encoder/model.safetensors",
                f"unet/diffusion_pytorch_model.safetensors",
        ]

Reply:
Hmm currently the behaviour on main is to return all the files in that list (both bin and safetensors) as  `usable_filenames` and I think the ignore patterns would remove the bin files, resulting in just the safetensors being downloaded.  

With this change only the fp16.bin files would be downloaded. Which feels technically "correct" to me since they are the "variant" files of each component. IMO non-variants should only be downloaded if no variant exists (regardless of format)

But this case implies that the proposal here is a breaking change, so I'll update to account for it.